python/pylibcudf/pylibcudf/io/parquet_metadata.pyx

```diff
@@ -0,0 +1,207 @@
+        """
+        return ParquetColumnSchema.from_column_schema(self.column_schema.child(idx))
+
+    cpdef list children(self):
+        """
+        Returns schemas of all child columns.
+
+        Returns
+        -------
+        list[ParquetColumnSchema]
+            Children schemas.
+        """
+        cdef cpp_parquet_metadata.parquet_column_schema child
+        return [
+            ParquetColumnSchema.from_column_schema(child)
+            for child in self.column_schema.children()
+        ]
+
+
+cdef class ParquetSchema:
+    """
+    Schema of a parquet file.
+
+    Parameters
+    ----------
+    parquet_schema
+    """
+
+    def __init__(self):
+        raise ValueError("Construct ParquetSchema with from_schema.")
+
+    @staticmethod
+    cdef from_schema(cpp_parquet_metadata.parquet_schema schema):
+        cdef ParquetSchema result = ParquetSchema.__new__(ParquetSchema)
+        result.schema = schema
+        return result
+
+    cpdef ParquetColumnSchema root(self):
+        """
+        Returns the schema of the struct column that contains all columns as fields.
+
+        Returns
+        -------
+        ParquetColumnSchema
+            Root column schema
+        """
+        return ParquetColumnSchema.from_column_schema(self.schema.root())
+
+
+cdef class ParquetMetadata:
+    """
+    Information about content of a parquet file.
+
+    Parameters
+    ----------
+    parquet_metadata
+    """
+
+    def __init__(self):
+        raise ValueError("Construct ParquetMetadata with from_metadata.")
+
+    @staticmethod
+    cdef from_metadata(cpp_parquet_metadata.parquet_metadata meta):
+        cdef ParquetMetadata result = ParquetMetadata.__new__(ParquetMetadata)
+        result.meta = meta
+        return result
+
+    cpdef ParquetSchema schema(self):
+        """
+        Returns the parquet schema.
+
+        Returns
+        -------
+        ParquetSchema
+            Parquet schema
+        """
+        return ParquetSchema.from_schema(self.meta.schema())
+
+    cpdef int num_rows(self):
+        """
+        Returns the number of rows of the root column.
+
+        Returns
+        -------
+        int
+            Number of rows
+        """
+        return self.meta.num_rows()
+
+    cpdef int num_rowgroups(self):
+        """
+        Returns the number of rowgroups in the file.
+
+        Returns
+        -------
+        int
+            Number of row groups.
+        """
+        return self.meta.num_rowgroups()
+
+    cpdef dict metadata(self):
+        """
+        Returns the Key value metadata in the file footer.
+
+        Returns
+        -------
+        dict[bytes, bytes]
+            Key value metadata as a map.
+        """
+        return self.meta.metadata()
+
+    cpdef list rowgroup_metadata(self):
+        """
+        Returns the row group metadata in the file footer.
+
+        Returns
+        -------
+        list[dict[str, int]]
+            Vector of row group metadata as maps.
```

Comment:
*nit*: why are we decoding the rowgroup metadata but not the metadata keys?

Reply:
That's fair. To be more Python-like I'll also decide the metadata keys